src/main/kotlin/org/rust/lang/core/psi/RsPsiFactory.kt

```diff
@@ -271,8 +271,8 @@ class RsPsiFactory(private val project: Project) {
             ?: error("Failed to create parameter element")
     }
 
-    fun createPatBinding(name: String, mutable: Boolean = false): RsPatBinding =
-        (createStatement("let ${if (mutable) "mut " else ""}$name = 10;") as RsLetDecl).pat
+    fun createPatBinding(name: String, mutable: Boolean = false, ref: Boolean = false): RsPatBinding =
+        (createStatement("let ${if (ref) "ref " else ""}${if (mutable) "mut " else ""}$name = 10;") as RsLetDecl).pat
```

Comment:
Maybe extract `if` expressions into separate variables to make the code more readable?